File: samples/cext-trey-research/src/model/dbModel.ts
import { TableEntity } from "@azure/data-tables";
import { Project, Consultant, Assignment } from "./baseModel";

export interface DbEntity extends TableEntity {
    etag: string;
    partitionKey: string;
    rowKey: string;
    timestamp: Date;
}

export interface DbProject extends DbEntity, Project { }

export interface DbConsultant extends DbEntity, Consultant { } 

export interface DbAssignment extends DbEntity, Assignment { }

File: samples/cext-trey-research-python/model/db_model.py
from dataclasses import dataclass, field
from datetime import datetime
from model.base_model import Project, Consultant, Assignment

@dataclass
class DbEntity:
    partition_key: str = field(default="")
    row_key: str = field(default="")
    etag: str = field(default="")
    timestamp: datetime = field(default_factory=datetime.utcnow)

@dataclass
class DbProject(DbEntity, Project):
    pass

@dataclass
class DbConsultant(DbEntity, Consultant):
    pass

@dataclass
class DbAssignment(DbEntity, Assignment):
    pass